win32/src/threadpool_win32.c

```diff
@@ -401,17 +311,8 @@ void threadpool_destroy_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WOR
         THREADPOOL* threadpool_ptr = THANDLE_GET_T(THREADPOOL)(threadpool);
 
         (void)InterlockedIncrement(&threadpool_ptr->pending_api_calls);
-
-        THREADPOOL_WIN32_STATE state = InterlockedAdd(&threadpool_ptr->state, 0);
-        if (state != THREADPOOL_WIN32_STATE_OPEN)
-        {
-            LogWarning("Bad state: %" PRI_MU_ENUM, MU_ENUM_VALUE(THREADPOOL_WIN32_STATE, state));
-        }
-        else
-        {
-            /* Codes_SRS_THREADPOOL_WIN32_05_016: [ threadpool_destroy_work_item shall call CloseThreadpoolWork to close ptp_work. ]*/
-            CloseThreadpoolWork(work_item_context->ptp_work);
-        }
+        /* Codes_SRS_THREADPOOL_WIN32_05_016: [ threadpool_destroy_work_item shall call CloseThreadpoolWork to close ptp_work. ]*/

```

Comment:
I fail to see why do we have pending_api_calls if there is no open/close state any longer.
I think we only wait for pending call count to be 0 in the dispose.
And if we are in the dispose, everyone has relinquished all the references.

So I think this pending_api_calls has to die.

Reply:
`pending_api_calls` is incremented by 1 just  before `SubmitThreadpoolWork` on line 290 which may happen in a different thread than where `threadpool_dispose` is happening. This ensures that perhaps the last few jobs to  `threadpool_schedule_work_item` is submitted before allow dispose to continue and destroy the `threadpool`. Not full proof but some kind of protection before dispose closes everything. Nothing to do with open and close.

Reply:
I beg to differ. Knowing the history from the first line of the code being written in the project, I can 100% say that pending_api_calls exists as a reference count so that we do not switch the state to closed while the count is greater than 0.

Can you prove your statement that you can call SubmitThreadpoolWork from one thread and dispose is being executed on another thread?

Reply:
Okay, I see what you are saying. But what is the harm in keeping them? Wouldn't it be useful if we want to bring back some feature like that in the future? Would it be helpful during debugs to see how many API calls were pending when some crash happened?

Reply:
Thanks for the clarification, I have updated the code.